### DISCUSSION TITLE
Windows Events

### DISCUSSION THREAD

**Discussion by paulobreim** [sepinf-inc/IPED#1385] [Q&A]
Does anyone know if it is possible to search for windows events?
For example, knowing all system boot dates, shutdown, user login, last run programs, etc.

tks,
Paulo

**Comment by lfcnassif** [comment]
@patrickdalla is improving evtx parsing. But many events you are interested in can be found in already existing registry reports, look at Windows Registry category for files with `-Report` suffix.

**Reply by paulobreim** [reply]
In fact a lot I saw in the windows registry.
Lately I've been trying to better understand what happens inside windows, and many things are strange, such as the creation date of a file being later than the modification date.

And inside the events, if we look at the Successful Logon(Event ID 4624) preset, for example, we will see that there are a lot of repetitions (10, 15) with differences of seconds, which is a little strange.

So, considering that most of the use of IPED is to attend forensic investigations, it would be interesting to have all the events listed, being able to create in the meta, a group of events and in the properties the different types of events.

Unfortunately I don't play with Java and I don't like it, I prefer C and lately I've been using Dart a lot, but I'm always available to help.

I'll see if I can find good documentation about the layout of events to try to do something interesting.

tks @lfcnassif

**Reply by lfcnassif** [reply]
> such as the creation date of a file being later than the modification date.

Probably the file was copied from another volume/partition/disk, then just the creation date is updated.

> it would be interesting to have all the events listed, being able to create in the meta, a group of events and in the properties the different types of events.

This already exists, look at the `Basic` properties group, you can find the `timeEvent` property, with all currently supported event types listed. They should increase in future versions of the tool.

**Comment by patrickdalla** [comment]
It has a functional "version" on Evtxparser branch.

The improvement remaining that i am working will be to name the eventIds.
The corresponding resource files with the evtx strings are registered in
the System registry. But they are more long explanatory strings. Maybe, for
a short name good enough to show in the chart, we must keep a separate
resource file.

But you could test the current version that extract timestamps of group of
records based on eventid, or even a parser that extracts each record as a
subitem.

Em dom., 16 de out. de 2022 21:39, Luis Filipe Nassif <
***@***.***> escreveu:

**Reply by paulobreim** [reply]
hi @patrickdalla,
can you tell me how I can test Evtxparser ?
tks

**Comment by paulobreim** [comment]
I saw this timeevent option that @lfcnassif quoted, excellent.
In the case of windows events, because it is a specific item, different from other OSs, I thought of something like presented in windows itself. There are security events, installation, applications, as well as presets like logon/logoff, shutdown/restart, successful and failed updates, critical events, and so on.
So I imagined a specific group just for events.
Additionally, we could include events of the type, USB events, when something was introduced in the USB like pendrive, hd, with date, equipment serial number, etc. These are things that you in the regripper that by altering the python a little we could create groups more used in skill.

**Reply by lfcnassif** [reply]
> a specific group just for events

We may create a specific metadata group including all Date properties/events, but this would move some of them from their current group, like `Image `for EXIF Dates or `Message` for email or instant message Dates, this could be bad for some kind of analysis...

**Reply by paulobreim** [reply]
In principle I agree.
I need to think a little more and wait for the final release of the version with "timeline".
I see the issue of windows events, as something separate from everything else because they are audit trails, such as: what dates did a certain user log on and log off, what dates the computer shut down, boot, entered sleep mode, etc.
